### ISSUE TITLE
Running synthea multiple times with same seed yields different results

### ISSUE THREAD

**Issue by tjones-hvn** [synthetichealth/synthea#752]
Hi, I'm running synthea specifying a seed, and I'm getting different results in `/output` when I'd expect to get identical results multiple times. 
I installed the .jar Synthea file for the first time last week. I'm following the Basic Setup and Running instructions, and the [wiki](https://github.com/synthetichealth/synthea/wiki/Basic-Setup-and-Running) says: `Populations generated with the same seed and the same version of Synthea should be identical`.

Here's what I'm running (note that I'm not specifying a custom `synthea.properties` file):
```
➜ java -jar synthea-with-dependencies.jar -s 9 -p 2
... 
Running with options:
Population: 2
Seed: 9
Provider Seed:1594648111560
Location: Massachusetts
Min Age: 0
Max Age: 140
1 -- Carina240 Hoppe518 (34 y/o F) Boston, Massachusetts
2 -- Perry780 Thompson596 (38 y/o F) Mashpee, Massachusetts
Records: total=2, alive=2, dead=0

➜ java -jar synthea-with-dependencies.jar -s 9 -p 2
...
Running with options:
Population: 2
Seed: 9
Provider Seed:1594648126542
Location: Massachusetts
Min Age: 0
Max Age: 140
1 -- Carina240 Fadel536 (34 y/o F) Boston, Massachusetts
2 -- Perry780 Gusikowski974 (38 y/o F) Mashpee, Massachusetts
Records: total=2, alive=2, dead=0

```

I would like to be able to run Synthea multiple times and get the exact same output files in `/output`. (As you can see from the output above, the last names are different). I've tried specifying a providerSeed as well and that also doesn't seem to help. How can I run Synthea multiple times and get the same result?

Thanks!

**Comment by jawalonoski** [comment]
Other than the last names, is any of the data in the records different?

There is a known issue where date/times in the records may be different, but I'm asking about diagnosed conditions, medications, etc.

Regarding the surnames, my guess based on the age and gender is that each woman got married and ended up with a different spouse... the maiden name should be the same.

This is still an issue though. We'll have to track it down.

**Comment by tjones-hvn** [comment]
Thanks for the prompt response. Yes, the data in the records is different too. This one example was easy to find, but I suspect there are many other differences.
`Carina240 Hoppe518` has three `Observation` resources where `Body temperature` was recorded, with values `37.704`, `37.719`, and `41.233`.
`Carina240 Fadel536` has just two `Observation` resources where `Body temperature` was recorded, with values `37.813` and `41.217`.

**Comment by jawalonoski** [comment]
Working on this. There are a few issues related to the use of `Random` throughout the code that will help. Right now I'm stuck with a multi-threading issue. If I generate each patient in serial, this issue goes away. If I generate them in parallel, this happens. So there is something that is happening in an unpredictable order.... attempting to track it down.

**Comment by tjones-hvn** [comment]
Thanks for working on this!

**Comment by jawalonoski** [comment]
This should be fixed in master since #756 was merged. We'll probably release a new version soon that includes this, since it is a pretty big fix.

**Comment by tjones-hvn** [comment]
Awesome! Thank you!
